### PR TITLE
Test a new implementation of calendar who's faster

### PR DESCRIPTION
**Pull request by Gwen678**
### **PR Description**
This update introduces the EventsPerDayScreen component, which dynamically displays events (classes, assignments, and tasks) on a daily calendar in a scrollable view. Key features include:

Event Fetching: Events are fetched from Firebase, including classes, assignments, and tasks specific to the authenticated user.
Dynamic Date Calculation: Generates a weekly calendar and dynamically calculates each day's date.
Event Rendering: Events are displayed with accurate start and end times, formatted in a user-friendly way. Events are sorted and displayed on a timeline that adjusts based on the user's current time.
Responsive Design: Supports portrait and landscape modes, with layout adjustments based on screen orientation.
Error Handling: Alerts users if there is an issue fetching events from Firebase.
Optimized Event Blocks: Events are displayed in 80px blocks with height adjustments based on their duration.
This component significantly improves the user experience by providing a clear and structured view of the day's and week's events, improving the ability to manage time effectively.
I want to do an other PR to put my test and improve score of sonarcloud...

Image 1
Image 2
Image 3
Image 4

### CODE REVIEW

File: edweiss-app/model/school/Events.ts
Comment (adamm-alaoui-pro) [inline edweiss-app/model/school/Events.ts:1]: I like the way you abstract away the events. This is a very good practice of creating an `interface` for this extent.
Maybe as a small improvement, you could create a `type CustumEventType = "Todo" | "Course" | "Assignment";`.

**Review by adamm-alaoui-pro — APPROVED** [review]
Before hand I would like to stress out the work you involved into getting this working much more smoothly.

I tested out your new version. It renders much faster than the previous iterations. You find an elegant way of tackling the rendering without significant lost of performance. As you may have noticed. I am a light mode crusador. I was first testing your app in dark mode to see how did it fit for your standpoint while developing it. Then I switch back to light mode. I was priorly a bit concerned about the `blanc cassé` that you used, it turns out it ended well. I also saw that you used a `flimongo` color. I would have just a small suggestion for later UI enhancement, maybe have a look at myy `LangSelModal`, it would be nice to make an alpha difference in the boxes so that the content can be seen in a more visible fashion. Beside this small suggestion, the swipping on portrait in working super well! However in landscape more it seems to be blocked (is it a bug or a feature 😁). Moreover, I like the way you handled the todo list access. It fits much better in the calendar. Time related stuff should be accessible from it. I tested out to access the todo list from the recorded ones, it works very well, great job at integrating it.

Regarding general comments, I would like to say even if it's hard to accept it, that you made a great choice of deleting your previous tests and version. It might be difficult to separate from almost a semester work.

Furthermore, I appreciate the modularization effort to separate the `VerticalCalendar` from the `horizontalCalendar`. That's makes things much simpler.

I did some small remarks to enhance the performances, you could leverage the memoization to avoid refetching everything from firebase when we rotate the screen. 

In addition to this, I see that you did not added new tests as sonar cloud is complaining about it. As you mentionned it at the end of your PR, in later iteration that would be great if you could fill this gap.

File: edweiss-app/app/(app)/calendar/index.tsx [hunk 1]
```diff
@@ -1,169 +1,162 @@
...
-    setPages((prevPages) => [...prevPages, prevPages[prevPages.length - 1] + 1]);
-  };
-
-  const getDateWithOffset = (offset: number): Date => {
-    const date = new Date();
-    date.setDate(date.getDate() + offset);
-    return date;
+      setEventsByDate(allEvents); // Update state with fetched events
+    } catch (error) {
+      Alert.alert('Error', "Unable to load events"); // Show error alert
```
[the remark this answers is not in the document]
Comment (Gwen678) [reply edweiss-app/app/(app)/calendar/index.tsx:129]: good idea i will implement it in new PR

File: edweiss-app/model/school/Events.ts
Comment (Gwen678) [reply edweiss-app/model/school/Events.ts:1]: good idea thanks

File: edweiss-app/components/core/getWeekDates.tsx
[the remark this answers is not in the document]
Comment (Gwen678) [reply edweiss-app/components/core/getWeekDates.tsx:1]: 👍

File: edweiss-app/app/(app)/calendar/VerticalCalendar.tsx [hunk 2]
```diff
@@ -0,0 +1,247 @@
...
+                                                </>
+                                            ) : (
+                                                <>
+                                                    {/* Display only startTime with a custom label */}
+                                                    <TText size='xs' color='text'>
+                                                        Scheduled at: {formatTime(event.startTime)}
+                                                    </TText>
+
+                                                </>
+                                            )}
```
[the remark this answers is not in the document]
Comment (Gwen678) [reply edweiss-app/app/(app)/calendar/VerticalCalendar.tsx:173]: ok i will check

File: edweiss-app/__test__/home_page/calendar.test.tsx
[the remark this answers is not in the document]
Comment (Gwen678) [reply edweiss-app/__test__/home_page/calendar.test.tsx:1]: yes thanks you see my work...

File: edweiss-app/app/(app)/calendar/VerticalCalendar.tsx [hunk 3]
```diff
@@ -0,0 +1,247 @@
...
+    useEffect(() => {
+        // Automatically scroll to the current time on component mount
+        scrollViewRef.current?.scrollTo({ y: (currentMinutes / 60 - 1) * HOUR_BLOCK_HEIGHT, animated: true });
+    }, [currentMinutes]);
+
+    useEffect(() => {
+        // Update the current time every minute
+        const interval = setInterval(() => setCurrentMinutes(getCurrentTimeInMinutes()), 60000);
+        return () => clearInterval(interval);
+    }, []);
```
[the remark this answers is not in the document]
Comment (Gwen678) [reply edweiss-app/app/(app)/calendar/VerticalCalendar.tsx:50]: i keep this idea for next sprint !

File: edweiss-app/app/(app)/calendar/VerticalCalendar.tsx (hunk 2 above)
[the remark this answers is not in the document]
Comment (adamm-alaoui-pro) [reply edweiss-app/app/(app)/calendar/VerticalCalendar.tsx:173]: Great!

File: edweiss-app/model/school/Events.ts
Comment (adamm-alaoui-pro) [reply edweiss-app/model/school/Events.ts:1]: Happy you like it.

**Pull request merged**